### PR TITLE
chore: abstract PropertyFilters logic

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

Towards https://github.com/PostHog/posthog/issues/22054#issuecomment-2139714014

Since https://github.com/PostHog/posthog/pull/22592 it seems silly to mess with `PropertyFilters`

Instead built a new concept of `UniversalFilters` in Replay (https://github.com/PostHog/posthog/pull/22612) that allows us to combine `PropertyFilters` and `ActionFilters`

Pulling the necessary changes for Product Analytics into a separate PR so that it's easier for them to review

## Changes

- Removed reliance on mounted logic
- Made `filters` and `setFilter` required props
- Allow `ReactNode` as children to the property filter button (prev. only allowed customised string)

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Updated existing tests

### CODE REVIEW

**Review by marandaneto — APPROVED** [review]

File: cypress/productAnalytics/index.ts [hunk 1]
```diff
@@ -173,7 +173,7 @@ export const dashboard = {
         cy.get('.PropertyFilterButton').should('have.length', 0)
         cy.get('[data-attr="property-filter-0"]').click()
         cy.get('[data-attr="taxonomic-filter-searchfield"]').click().type('Browser').wait(1000)
-        cy.get('[data-attr="prop-filter-event_properties-0"]').click({ force: true })
+        cy.get('[data-attr="prop-filter-event_properties-0"]').click({ force: true }).wait(1000)
```
Comment (daibhin) [inline cypress/productAnalytics/index.ts:176]: Test was flapping because the previous input hadn't been torn down quick enough. Unclear why this change slowed things down but 🤷
